contracts/pcv/maple/MaplePCVDeposit.sol

```diff
@@ -0,0 +1,354 @@
+        mapleRewards = _mapleRewards;
+        mapleToken = IMapleRewards(_mapleRewards).rewardsToken();
+        pcvOracle = _pcvOracle;
+    }
+
+    /// @notice return the amount of funds this contract owns in USDC
+    /// accounting for interest earned
+    /// and unrealized losses in the venue
+    function balance() public view override returns (uint256) {
+        return
+            IMaplePool(maplePool).balanceOf(address(this)) /
+            SCALING_FACTOR +
+            IMaplePool(maplePool).accumulativeFundsOf(address(this)) -
+            IMaplePool(maplePool).recognizableLossesOf(address(this));
+    }
+
+    /// @notice return the underlying token denomination for this deposit
+    function balanceReportedIn() external view returns (address) {
+        return USDC;
+    }
+
+    /// @notice set the pcv oracle address
+    /// @param _pcvOracle new pcv oracle to reference
+    function setPCVOracle(address _pcvOracle) external onlyGovernor {
+        address oldOracle = pcvOracle;
+        pcvOracle = _pcvOracle;
+
+        _recordPNL();
+
+        IPCVOracle(pcvOracle).updateIlliquidBalance(
+            lastRecordedBalance.toInt256()
+        );
+
+        emit PCVOracleUpdated(oldOracle, _pcvOracle);
+    }
+
+    /// ---------- Happy Path APIs ----------
+
+    /// @notice deposit PCV into Maple.
+    /// deposits are subject to up to 30 days of lockup,
+    /// weighted by the amount deposited and the deposit time.
+    /// No op if 0 token balance.
+    /// Deposits are then immediately staked to accrue MPL rewards.
+    /// Only pcv controller can deposit, as this contract would be vulnerable
+    /// to donation / griefing attacks if anyone could call deposit and extend lockup time.
+    function deposit() external onlyPCVController {
+        uint256 amount = IERC20(USDC).balanceOf(address(this));
+        if (amount == 0) {
+            /// no op to prevent wasted gas
+            return;
+        }
+
+        int256 startingRecordedBalance = lastRecordedBalance.toInt256();
+
+        /// ------ Effects ------
+
+        /// compute profit from interest accrued and emit an event
+        /// if any profits or losses are realized
+        _recordPNL();
+
+        lastRecordedBalance += amount;
+
+        /// ------ Interactions ------
+
+        /// pool deposit
+        IERC20(USDC).approve(maplePool, amount);
+        IMaplePool(maplePool).deposit(amount);
+
+        /// stake pool FDT for MPL rewards
+        uint256 scaledDepositAmount = amount * SCALING_FACTOR;
+        IMaplePool(maplePool).increaseCustodyAllowance(
+            mapleRewards,
+            scaledDepositAmount
+        );
+        IMapleRewards(mapleRewards).stake(scaledDepositAmount);
+
+        int256 endingRecordedBalance = lastRecordedBalance.toInt256();
+
+        if (pcvOracle != address(0)) {
+            IPCVOracle(pcvOracle).updateIlliquidBalance(
+                endingRecordedBalance - startingRecordedBalance
```

Comment:
Due to this math in the contract when calling `updateIlliquidBalance`, I'd be in favor of adding an invariant test that's quite similar to the MorphoPCVDeposit invariants, that asserts this contract reports values correctly.